app/controllers/concerns/hal_actions.rb

```diff
@@ -16,14 +16,14 @@ def show
   end
 
   def create
-    self.resource = consume!(resource)
+    consume_with_content_type! resource
```

Comment:
I'm closing this PR because it's included in the refactor, but this works because `consume!` transforms the resource in place.